### PR TITLE
Ribbon client default properties

### PR DESCRIPTION
**Pull request by sabareeshkkanan**
PropertiesFactory will load default config for all client from propeties unless overridden by the client specific properties

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@sabareeshkkanan Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud/spring-cloud-netflix&pullRequestId=3576)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-cloud/spring-cloud-netflix&pullRequestId=3576) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@sabareeshkkanan Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud/spring-cloud-netflix&pullRequestId=3576)!

File: spring-cloud-netflix-ribbon/src/main/java/org/springframework/cloud/netflix/ribbon/PropertiesFactory.java [hunk 1]
```diff
@@ -59,6 +60,9 @@ public String getClassName(Class clazz, String name) {
 			String classNameProperty = this.classToProperty.get(clazz);
 			String className = environment
 					.getProperty(name + "." + NAMESPACE + "." + classNameProperty);
+			if(className==null)
```
Comment (spencergibb) [inline spring-cloud-netflix-ribbon/src/main/java/org/springframework/cloud/netflix/ribbon/PropertiesFactory.java]: Please use curly braces for if statements

**Comment by sabareeshkkanan** [comment]
@mbronshteyn can you write test for this

**Comment by sabareeshkkanan** [comment]
will recreate

**Pull request closed without merging**